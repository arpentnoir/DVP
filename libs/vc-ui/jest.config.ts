/* eslint-disable */
export default {
  displayName: 'vc-ui',
  preset: '../../jest.preset.js',
  setupFiles: ['./jest.setup.ts'],
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/vc-ui',
  moduleNameMapper: {
    '^react-pdf': 'react-pdf/dist/umd/entry.jest',
  },
};
