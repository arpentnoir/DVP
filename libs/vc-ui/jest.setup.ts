import 'jest-canvas-mock';
import '@testing-library/jest-dom';

const mockClipboard = {
  writeText: jest.fn(),
};

(global.navigator.clipboard as any) = mockClipboard;
