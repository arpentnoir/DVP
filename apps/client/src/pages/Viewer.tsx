import { Box } from '@mui/material';
import { IStatusCheck, VerifyViewer, VALID_CHECKS } from '@dvp/vc-ui';
import { useLocation, useNavigate } from 'react-router-dom';
import { useEffect, useState } from 'react';
import { ROUTES } from '../constants';

export const Viewer = () => {
  const [checks, setChecks] = useState<IStatusCheck[]>([]);
  const location = useLocation();
  const navigate = useNavigate();
  const { state } = location;

  useEffect(() => {
    // Redirect to home page if state is not available
    if (!state) {
      navigate(ROUTES.HOME);
    } else {
      if (!state.results.errors.length) {
        setChecks(VALID_CHECKS);
      }
    }
  }, []);

  return state ? (
    <Box paddingTop={6} position={'relative'}>
      <VerifyViewer
        document={state.document}
        results={{ ...state.results, checks }}
        hideVerifyResults={state.hideVerifyResults}
      />
    </Box>
  ) : null;
};
