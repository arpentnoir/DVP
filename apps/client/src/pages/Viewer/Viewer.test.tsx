import { render } from '@dvp/vc-ui';
import { Viewer } from './Viewer';

describe('Viewer', () => {
  it('should render correctly', () => {
    const { baseElement } = render(<Viewer />);

    expect(baseElement).toMatchSnapshot();
  });
});
