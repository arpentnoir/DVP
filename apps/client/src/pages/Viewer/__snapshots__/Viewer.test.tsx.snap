// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viewer should render correctly 1`] = `
<body>
  <div />
</body>
`;
